src/cript/api/exceptions.py

```diff
@@ -110,7 +111,7 @@ class CRIPTAPISaveError(CRIPTException):
     http_code: str
     api_response: str
 
-    def __init__(self, api_host_domain: str, http_code: str, api_response: str, patch_request: bool, pre_saved_nodes: Optional[Set[str]] = None, json_data: Optional[str] = None):
+    def __init__(self, api_host_domain: str, http_code: str, api_response: str, patch_request: bool, pre_saved_nodes: Set[str], json_data: str):
```

Comment:
I am okay approving this, but just concerned that if our error messages get too big it will be really hard to read and debug for users and ourselves.

We can also tackle cleaning up our errors after we have all features done

Reply:
The error message doesn't actually get longer.
The contrary, the message itself is shorter now.

There is a fundamental reason for requiring all this information.
This extra info can be used to clean up save calls that go wrong.
For example we have the infrastructure to delete trees that got only saved halfway through.
We were just waiting for `delete` to implement it.

And by inheriting and passing this info down, this capability remains intact even for NameErrors during saving, but with an improved error message.